### PR TITLE
Update dimradial draw function

### PR DESCRIPTION
**Pull request by bhattigurjot**

### CODE REVIEW

File: lcviewernoqt/drawitems/lcdimradial.cpp [hunk 1]
```diff
@@ -34,21 +34,67 @@ void LCDimRadial::draw(LcPainter& painter, const LcDrawOptions &options, const l
...
+                    ? lc::geo::Coordinate(mousePos.x() - (te.width/2 + capSize/2), mousePos.y())
+                    : lc::geo::Coordinate(mousePos.x() + (te.width/2 + capSize/2), mousePos.y());
+        }
+
+        painter.move_to(definitionPoint2().x(), definitionPoint2().y());
+        painter.line_to(mousePos.x(), mousePos.y());
+        painter.line_to(tLinePos.x(), mousePos.y());
+        painter.stroke();
+        endCaps.render(painter, EndCaps::CLOSEDARROW, mousePos.x(), mousePos.y(), definitionPoint2().x(), definitionPoint2().y(), capSize) ;
+
```
Comment (dxli) [inline lcviewernoqt/drawitems/lcdimradial.cpp]: Trying to cleanup a little bit here, so the code reads easier:

```C++
    using namespace lc::geo;

    Coordinate offsetX{te.width + capSize, 0};
    Coordinate offsetY{0, te.width/4 + capSize/2};

    if (textisInside && textFitsInside) {
	auto const offset = (definitionPoint2().x() > mousePos.x()) ? - offsetX : offsetX;
	auto const tLinePos = mousePos + offset;
	auto const tMText = mousePos + offsetX/2;

	if  (definitionPoint2().y() > mousePos.y()) {
		tMText -= offsetY;
	}

	//the painter code might read better with some adaptive interface
	auto move_to = [&painter] (Coordinate const& coord) {painter.move_to(coord.x, coord.y);}
	auto line_to = [&painter] (Coordinate const& coord) {painter.line_to(coord.x, coord.y);}

        move_to(definitionPoint2());
        line_to(mousePos);

```

the logic can be further simplified as far as the intention is still clear enough to be read from code.

Comment (rvt) [inline lcviewernoqt/drawitems/lcdimradial.cpp]: dli,

perhaps we can create some header for that adaptive interface with some generic (wildly guessing).
I wanted to keep our entities (Coordinate etc... etc) out of the painter functions so we don't have potential circular references, or at a minimum that the painter doesn't depend on the geo classes..

Comment (dxli) [inline lcviewernoqt/drawitems/lcdimradial.cpp]: I agree.

mixing geometry/logic and dependency of painter interface seems to be bad.

Comment (bhattigurjot) [inline lcviewernoqt/drawitems/lcdimradial.cpp]: Do you want me to make such changes?

Comment (dxli) [inline lcviewernoqt/drawitems/lcdimradial.cpp]: @bhattigurjot 
please make your own judgement call, and avoid get into the "deadlock" situation, as I explained early on: while one party is waiting for comments, and the other party is waiting to see actions.

Thanks!

**Pull request merged**